great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -491,11 +494,15 @@ def add_datasource(
         else:
             config = kwargs
 
+        datasource_config: DatasourceConfig = datasourceConfigSchema.load(
+            CommentedMap(**config)
+        )
```

Comment:
Is there a reason we use `CommentedMap` here?

Reply:
I believe this supports comments in yaml, I left it in to support existing functionality

Reply:
works for me! I'd rather not expose `ruamel` internals outside of `YAMLHandler` but this isn't in scope for your PR